alerts/osdsdown.libsonnet

```diff
@@ -0,0 +1,24 @@
+{
+  prometheusAlerts+:: {
+    groups+: [
+      {
+        name: 'osds-down-alert.rules',
+        rules: [
+          {
+            alert: 'CephOSDsDown',
+            expr: |||
+              count(ceph_osd_metadata) - count(ceph_osd_up) > 0.5
```

Comment:
All you need here is `ceph_osd_up < 0.5`.

Reply:
Ok. Will try that.

Reply:
Not sure what the intended environment is - but for a hands-off k8s environment raising an alert for down OSDs is probably not what we want - OSDs can sit in that state indefinitely. So perhaps we should look for down+out and recovery is active to form the basis of an alert?

Reply:
Would something like this work - (count(ceph_osd_up * on(ceph_daemon) ceph_osd_in == 0) > 0) and ON() ceph_pg_total - ceph_pg_clean > 0

Reply:
Will give a try.
Verification is something I find really tough in k8s environment though.

Reply:
can you cause the pod to crash loop by detaching the disk?

Reply:
I am running the setup on VMs and OSDs created out of OS disk itself.
May be in a standalone setup where OSDs created out of specific disks, I should ne able to test it out. Will try the same.